### ISSUE TITLE
GrantPermissions issue (ios)

### ISSUE THREAD

**Issue by urlexnilz** [dpa99c/cordova-plugin-firebasex#199]
Dear all,

I successfully implemented FirebaseX. On start of my app I see a prompt for asking for permissions on ios to send push notifications. My assumption was that if the user clicks on "disallow" on next app start it will ask you again to allow / disallow push notifications, but I dont see any further prompts if once clicked on "disallow".  Is my assumption wrong?

2019-10-29 14:38:34.240568+0100 xyz-app[374:13121] requestAuthorizationWithOptions: granted=NO
2019-10-29 14:38:34.242321+0100 xyz-app[374:13092] requestAuthorizationWithOptions: granted=NO

	grantPermissions() {
		this.push.grantPermission().then(hasPermission => {
			console.log('Permission was ' + (hasPermission ? "granted" : "denied"));
		});
	}

**Comment by andysousa** [comment]
iOS doesn't allow you to prompt the user again... its a done deal on the first time.  The only way to turn notifications on after they say no is to do so in the settings of the phone. 

We put an info page up telling the user why we need push notifications... its important to make it clear so the user doesn't decline for that very reason.

**Comment by dpa99c** [comment]
@andysousa is correct - once a user presses "Disallow" there is no way to programmatically request permission again. 

You could use the [getRemoteNotificationsAuthorizationStatus()](https://github.com/dpa99c/cordova-diagnostic-plugin#getremotenotificationsauthorizationstatus) method from [cordova-diagnostic-plugin](https://github.com/dpa99c/cordova-diagnostic-plugin) to identify if the permission status is denied and if so display information to the user instructing them how to manually re-enable notifications via the Settings app.

**Comment by urlexnilz** [comment]
Yeah, thanks for the info. I solved it in a more accurate way (imo):
- Ask for permissions
- if granted: everything is fine
- if not: display AlertController and use cordova open native settings plugin (https://github.com/guyromb/Cordova-open-native-settings)

could improve it like: app is started for the 5th time.. display alert: 

```
	grantPermissions() {
		this.push.grantPermission().then(hasPermission => {
			if (!hasPermission) {
				// logic to display popup every x times				
				this.presentAlertConfirm().then(() => {
					console.log('alert controller presented');
				});
			}
			console.log('Permission was ' + (hasPermission ? "granted" : "denied"));
		});
	}
```

my alert controller:

	async presentAlertConfirm() {
		const alert = await this.alertController.create({
			header: 'Activate Push Notifications',
			message: 'your message',
			buttons: [
				{
					text: 'Ignore',
					role: 'cancel',
					cssClass: 'secondary',
					handler: (blah) => {
						console.log('ignored...');
					}
				}, {
					text: 'switch to Settings',
					handler: () => {
						this.nativeSettings.open('application_details').then(() => {
							console.log('opened native settings ( => application_details)');
						})
					}
				}
			]
		});
		
		await alert.present();
	}

**Comment by dpa99c** [comment]
That is of course fine and handles the case where they accidentally press Disallow
but doesn't handle the edge case where the user manually disables remote notifications via Settings. 
If remote notifications were critical to your app functionality, you'd want to check their authorization status at app startup/resume as well.

Diagnostic plugin also has an equivalent method to switch directly to the app settings: [switchToSettings()](https://github.com/dpa99c/cordova-diagnostic-plugin#switchtosettings)

**Comment by urlexnilz** [comment]
@dpa99c: you mean remote notifications for ALL apps? I just checked for my given app and it works fine 💯 (disabled it manually in settings under apps node).
